pkg/router/controller/host_admitter.go

```diff
@@ -246,7 +246,8 @@ func (p *HostAdmitter) displacedRoutes(newRoute *routev1.Route) ([]*routev1.Rout
 	// See if any existing routes block our host, or if we displace their host
 	for i, route := range p.claimedHosts[newRoute.Spec.Host] {
 		if p.disableNamespaceCheck || route.Namespace == newRoute.Namespace {
-			if !p.disableNamespaceCheck && route.Name == newRoute.Name {
+			if route.Namespace == newRoute.Namespace && route.Name == newRoute.Name {
```

Comment:
Is this intended to establish identity? If so, what happens if the routes are actually different resources (different UIDs) with the same namespace/name (e.g. in a delete/create scenario)?

Reply:
And if so, would a UID comparison do the job but without any of the ambiguity of checking just namespace/name

Reply:
That's true - a `uid` check would work better here. Will change it and check re: the other failing fuzzing test.